src/compiler/binder.ts

```diff
@@ -77,19 +92,26 @@ namespace ts {
         IsContainerWithLocals = IsContainer | HasLocals
     }
 
-    export function bindSourceFile(file: SourceFile) {
+    export function bindSourceFile(file: SourceFile, options: CompilerOptions) {

```

Comment:
Now that binding depends on compiler options, do we appropriately invalidate the computed binding state when the compiler options change?